processor/src/processor/operations/io_ops.rs

```diff
@@ -255,3 +257,109 @@ pub(super) fn op_pipe<P: Processor>(
 
     Ok(())
 }
+
+#[inline(always)]
+pub(super) fn op_crypto_stream<P: Processor>(
+    processor: &mut P,
+    err_ctx: &impl ErrorContext,
+    tracer: &mut impl Tracer,
+) -> Result<(), ExecutionError> {
+    /// WORD_SIZE, but as a `Felt`.
+    const WORD_SIZE_FELT: Felt = Felt::new(4);
+    /// The size of a double-word.
+    const DOUBLE_WORD_SIZE: Felt = Felt::new(8);
+
+    // Stack layout: [rate(8), capacity(4), src_ptr, dst_ptr, ...]
+    const SRC_PTR_IDX: usize = 12;
+    const DST_PTR_IDX: usize = 13;
+
+    let ctx = processor.system().ctx();
+    let clk = processor.system().clk();
+
+    // Get source and destination pointers
+    let src_addr = processor.stack().get(SRC_PTR_IDX);
+    let dst_addr = processor.stack().get(DST_PTR_IDX);
+
+    if src_addr == dst_addr {
```

Comment:
same comment as in the slow processor implementation

Reply:
How about doing a traditional overlap check?
```
  let src_end = src_addr + DOUBLE_WORD_SIZE;
  let dst_end = dst_addr + DOUBLE_WORD_SIZE;
  if src_addr < dst_end && dst_addr < src_end {
      return Err(ExecutionError::MemoryError(...));
  }
```
Note : the same issue is present in crypto_ops.

Whichever solution is adopted in the end, I'd highly advise unit tests covering this overlap case!

Reply:
Good point!
I think letting the memory chiplet handle concurrent accesses is the write solution but since we are doing 2 reads and 2 writes and hence the potential for overlap, and as this is specific only to this op, I think that adding checks at op level is the way to go.

Went ahead and encapsulated the checks into a helper function to aid readability, and also added some tests to catch overlaps.